lxd/db/generate/db/stmt.go

```diff
@@ -470,13 +502,17 @@ func whereClause(fields []*Field) string {
 
 	for i, field := range directFields {
 		if field.IsScalar() {
-			ref := lex.Snake(field.Name)
-			refTable := entityTable(ref, "")
-			subSelect := fmt.Sprintf("SELECT %s.id FROM %s", refTable, refTable)
+			joinTable, joinColumn, err := field.ScalarTableColumn()
+			if err != nil {
+				return "", err
```

Comment:
Maybe wrap the error with field name here?

Reply:
I think we can wrap the struct name here (and below) with `mapping.Name`, and include the field name in the function's own errors.

Reply:
Can do but the field name is in the `Field` struct already so may as well include it in the error string.

Reply:
OK.